### PR TITLE
Add support for accessing detailed test info via toMatchSnapshot callback

### PR DESCRIPTION
**Pull request by Swingline0**
Implements the proposed change agreed upon in #82 along with updates to relevant unit/integration tests and API.md.

@NimaSoroush  I _think_ this proposal should be fairly self-explanetory based on our previous discussion but please let me know if you have any questions or would like to see any revisions made before moving forward.

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -167,6 +167,88 @@ In this example, differencify will launch a browser instance and share same brow
 ```
 In this example, after calling `result` function it will return the previous step result as an object.
 
+## Detailed Result Information
```
Comment (NimaSoroush) [inline API.md]: You probably need to update [Additional methods on top of Puppeteer's Page class table](https://github.com/NimaSoroush/differencify/blob/master/API.md#additional-methods-on-top-of-puppeteers-page-class). It still says `toMatchSnapshot -> no argument`

toMatchSnapshot | no argument
-- | --




toMatchSnapshot | no argument
-- | --




toMatchSnapshot | no argument
-- | --




toMatchSnapshot | no argument
-- | --

**Comment by NimaSoroush** [comment]
Hi @Swingline0 , PR looks great. Just added a small comment.
You probably need to bump up version in package.json and add a changelog too.

**Comment by Swingline0** [comment]
@NimaSoroush I added the argument description via 491044f. Let me know if it reads correct from a technical perspective or needs revision.

Regarding the version bump, as this is affecting function arguments, do you wish to bump the minor?

**Comment by NimaSoroush** [comment]
Thanks @Swingline0, Looks great.

Yeah, a minor should be fine

**Comment by NimaSoroush** [comment]
Great. I'll merge then

**Pull request merged**